src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -217,22 +236,81 @@ private void handleStart(HttpExchange exchange) throws IOException {
         }
     }
 
+    private void handleStopOrUpdate(HttpExchange exchange, long id) throws IOException {
+        try {
+            JFRConnection conn =
+                    jfrConnectionToolkit.connect(
+                            jfrConnectionToolkit.createServiceURL("localhost", 0));
+            IFlightRecorderService svc = conn.getService();
+            IRecordingDescriptor dsc = svc.getAvailableRecordings().stream().filter(r -> r.getId() == id).findFirst().get();
+            RecordingOptionsBuilder builder = new RecordingOptionsBuilder(conn.getService());
+            
+            InputStream body = exchange.getRequestBody();
+            JsonNode jsonMap = mapper.readTree(body);
+            Iterator<Entry<String, JsonNode>> fields = jsonMap.fields();
+
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> field = fields.next();
+
+                switch(field.getKey()){
+                    case "state":
+                        if ("STOPPED".equals(field.getValue().toString())) {
+                            handleStop(exchange, id);
+                            break;
+                        } else if (!"UPDATE".equals(field.getValue().toString())) {
```

Comment:
Can this `state: "UPDATE"` case be removed? I think the only time the client needs to specify the `state` key currently is if they are trying to put the recording into `STOPPED` state - so if the value is anything other than `STOPPED`, it should result in a `BAD REQUEST`. If the client just wants to perform an update of other recording options without changing the state, they simply don't pass the `state` in the request.